packages/overlay/src/overlay-stack.ts

```diff
@@ -173,19 +173,23 @@ export class OverlayStack {
         }
 
         const activeOverlay = ActiveOverlay.create(details);
-        this.overlays.push(activeOverlay);
         document.body.appendChild(activeOverlay);
-        let updateComplete = await activeOverlay.updateComplete;
-        while (updateComplete === false) {
-            updateComplete = await activeOverlay.updateComplete;
-        }
 
-        this.addOverlayEventListeners(activeOverlay);
-        if (details.receivesFocus === 'auto') {
-            activeOverlay.focus();
-        }
+        return new Promise((res) => {
+            setTimeout(res);
+        }).then(async () => {
+            this.overlays.push(activeOverlay);
+            let updateComplete = await activeOverlay.updateComplete;
+            while (updateComplete === false) {
+                updateComplete = await activeOverlay.updateComplete;
+            }
```

Comment:
it's a little unfortunate you need 4 lines for this "wait until fully updated" check. It'd be nice if litElement had a utility. like `allPendingUpdatesComplete`

Reply:
This is less specifically `LitElement` than the way we acquire the overlay content after the first render in `ActiveOverlay`. I should be able to clean this up...